java.lang.NoClassDefFoundError: org/terracotta/entity/SyncMessageCodec

Is there a packaging issue with the latest ehcache dependency ? I have a project depending on:

```
<dependency>
  <groupId>org.ehcache</groupId>
  <artifactId>ehcache-clustered</artifactId>
  <version>3.2.0-SNAPSHOT</version>
</dependency>
```

When creating a clustered CM, I have the exception:

```
java.lang.NoClassDefFoundError: org/terracotta/entity/SyncMessageCodec
    at org.ehcache.clustered.client.internal.lock.VoltronReadWriteLockEntityClientService.getMessageCodec(VoltronReadWriteLockEntityClientService.java:52)
    at com.terracotta.connection.entity.TerracottaEntityRef.fetchEntity(TerracottaEntityRef.java:80)
    at org.ehcache.clustered.client.internal.lock.VoltronReadWriteLock.createClientEntity(VoltronReadWriteLock.java:125)
    at org.ehcache.clustered.client.internal.lock.VoltronReadWriteLock.tryLock(VoltronReadWriteLock.java:64)
    at org.ehcache.clustered.client.internal.lock.VoltronReadWriteLock.tryWriteLock(VoltronReadWriteLock.java:54)
    at org.ehcache.clustered.client.internal.EhcacheClientEntityFactory.create(EhcacheClientEntityFactory.java:98)
    at org.ehcache.clustered.client.internal.service.DefaultClusteringService.autoCreateEntity(DefaultClusteringService.java:198)
    at org.ehcache.clustered.client.internal.service.DefaultClusteringService.start(DefaultClusteringService.java:146)
    at org.ehcache.core.internal.service.ServiceLocator.startAllServices(ServiceLocator.java:118)
    at org.ehcache.core.EhcacheManager.init(EhcacheManager.java:560)
    at org.terracotta.websessions.store.ehcache3.Ehcache3SessionStore.doInit(Ehcache3SessionStore.java:64)
    at org.terracotta.websessions.api.AbstractSessionStore.init(AbstractSessionStore.java:69)
    at org.terracotta.websessions.support.tomcat8.TerracottaTomcatStore.initInternal(TerracottaTomcatStore.java:52)
    at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:107)
    at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:140)
    at org.apache.catalina.session.PersistentManagerBase.startInternal(PersistentManagerBase.java:847)
    at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
    at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5206)
    at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
    at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1403)
    at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1393)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassNotFoundException: org.terracotta.entity.SyncMessageCodec
    at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
    ... 25 more
```

--- a/clustered/common/build.gradle
+++ b/clustered/common/build.gradle
@@ -17,5 +17,5 @@
 apply plugin: EhDeploy
 
 dependencies {
-  provided "org.terracotta:entity-server-api:$parent.entityApiVersion"
+  provided "org.terracotta:entity-common-api:$parent.entityApiVersion"
 }

--- a/clustered/common/src/main/java/org/ehcache/clustered/common/internal/lock/LockMessaging.java
+++ b/clustered/common/src/main/java/org/ehcache/clustered/common/internal/lock/LockMessaging.java
@@ -20,7 +20,6 @@ import org.terracotta.entity.EntityMessage;
 import org.terracotta.entity.EntityResponse;
 import org.terracotta.entity.MessageCodec;
 import org.terracotta.entity.MessageCodecException;
-import org.terracotta.entity.SyncMessageCodec;
 
 public class LockMessaging {
 
@@ -60,24 +59,8 @@ public class LockMessaging {
     }
   };
 
-  private static final SyncMessageCodec<LockOperation> SYNC_CODEC = new SyncMessageCodec<LockOperation>() {
-    @Override
-    public byte[] encode(int i, LockOperation message) throws MessageCodecException {
-      throw new AssertionError();
-    }
-
-    @Override
-    public LockOperation decode(int i, byte[] bytes) throws MessageCodecException {
-      throw new AssertionError();
-    }
-  };
-
   public static MessageCodec<LockOperation, LockTransition> codec() {
     return CODEC;
-  }
-
-  public static SyncMessageCodec<LockOperation> syncCodec() {
-    return SYNC_CODEC;
   }
 
   public static LockOperation tryLock(HoldType type) {

--- a/clustered/server/src/main/java/org/ehcache/clustered/lock/server/VoltronReadWriteLockServerEntityService.java
+++ b/clustered/server/src/main/java/org/ehcache/clustered/lock/server/VoltronReadWriteLockServerEntityService.java
@@ -21,6 +21,7 @@ import org.ehcache.clustered.common.internal.lock.LockMessaging;
 import org.ehcache.clustered.common.internal.lock.LockMessaging.LockOperation;
 import org.ehcache.clustered.common.internal.lock.LockMessaging.LockTransition;
 
+import org.ehcache.clustered.lock.server.messages.LockSyncMessaging;
 import org.terracotta.entity.ActiveServerEntity;
 import org.terracotta.entity.ClientCommunicator;
 import org.terracotta.entity.ConcurrencyStrategy;
@@ -80,7 +81,7 @@ public class VoltronReadWriteLockServerEntityService implements EntityServerServ
 
   @Override
   public SyncMessageCodec<LockOperation> getSyncMessageCodec() {
-    return LockMessaging.syncCodec();
+    return LockSyncMessaging.syncCodec();
   }
 
   private static final <T> ServiceConfiguration<T> config(final Class<T> klazz) {

--- a/clustered/server/src/main/java/org/ehcache/clustered/lock/server/messages/LockSyncMessaging.java
+++ b/clustered/server/src/main/java/org/ehcache/clustered/lock/server/messages/LockSyncMessaging.java
@@ -1,0 +1,44 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.lock.server.messages;
+
+import org.ehcache.clustered.common.internal.lock.LockMessaging;
+import org.terracotta.entity.MessageCodecException;
+import org.terracotta.entity.SyncMessageCodec;
+
+/**
+ * LockSyncMessaging
+ */
+public class LockSyncMessaging {
+
+  public static SyncMessageCodec<LockMessaging.LockOperation> syncCodec() {
+    return SYNC_CODEC;
+  }
+
+  private static final SyncMessageCodec<LockMessaging.LockOperation> SYNC_CODEC = new SyncMessageCodec<LockMessaging.LockOperation>() {
+    @Override
+    public byte[] encode(int i, LockMessaging.LockOperation message) throws MessageCodecException {
+      throw new AssertionError();
+    }
+
+    @Override
+    public LockMessaging.LockOperation decode(int i, byte[] bytes) throws MessageCodecException {
+      throw new AssertionError();
+    }
+  };
+
+}

